apps/meteor/ee/server/apps/storage/AppRealStorage.ts

```diff
@@ -46,7 +47,18 @@ export class AppRealStorage extends AppMetadataStorage {
 	}
 
 	public async update(item: IAppStorageItem): Promise<IAppStorageItem> {
-		await this.db.updateOne({ id: item.id, _id: item._id }, { $set: item });
+		const updateQuery: UpdateFilter<IAppStorageItem> = {
+			$set: item,
+		};
+
+		// Note: This is really important, since we currently store the permissionsGranted as null if none are present
+		//       in the App's manifest. So, if there was a permissionGranted and it was removed, we must see the app as having
+		//       no permissionsGranted at all (which means default permissions). So we must actively unset the field.
+		if (!item.permissionsGranted) {
+			updateQuery.$unset = { permissionsGranted: 1 };
+		}
+
+		await this.db.updateOne({ id: item.id, _id: item._id }, updateQuery);
```

Comment:
Can we replace the `updateOne` method call with a `findOneAndUpdate`? Then we can save on DB network roundtrip, removing the query below

Reply:
yep